mParticle-Apple-SDK/Network/MPNetworkCommunication.m

```diff
@@ -209,6 +219,34 @@ - (MPURL *)eventURLForUpload:(MPUpload *)mpUpload {
     return eventURL;
 }
 
+- (MPURL *)audienceURL {
+    MPStateMachine *stateMachine = [MParticle sharedInstance].stateMachine;
+    
+    NSString *eventHost = [MParticle sharedInstance].networkOptions.eventsHost ?: self.defaultEventHost;
+    NSString *audienceURLFormat = [audienceFormat stringByAppendingString:@"?mpid=%@"];
+    NSString *urlString = [NSString stringWithFormat:audienceURLFormat, kMPURLScheme, self.defaultEventHost, kMPAudienceVersion, stateMachine.apiKey, kMPAudienceURL, [MPPersistenceController mpId]];
```

Comment:
this line is probably not needed since it's being set in 232/235

Reply:
OK so now that I'm looking at this myself I notice that on line 228 we are using this to create the default URL which is then used to create the MPURL on line 244 (which has a modified and default url within it) so it is needed

Reply:
thanks for pointing that out